dask/dataframe/groupby.py

```diff
@@ -530,6 +530,22 @@ def _finalize_std(df, count_column, sum_column, sum2_column, ddof=1):
     return np.sqrt(result)
 
 
+def _cum_agg_aligned(part, cum_last, index, columns, func, neutral):
+    align = cum_last.reindex(part.set_index(index).index, fill_value=neutral)
+    align.index = part.index
+    return func(part[columns], align)
+
+
+def _cum_agg_filled(a, b, func, neutral):
+    union = a.index.union(b.index)
+    return func(a.reindex(union, fill_value=neutral),
+                b.reindex(union, fill_value=neutral), fill_value=neutral)
```

Comment:
You're totally right. Another way would be to use `append` and then `groupby` again, but reindex approach feels more natural, perhaps it will just apply func when the groupings are the same for different partitions. `a.combine(b, func, fill_value=...)` could also be used, but dtype may change for dataframes with combine, so I did fall back to `reindex`.